Sources/DataPipeline/DataPipelineImplementation.swift

```diff
@@ -2,14 +2,33 @@ import CioInternalCommon
 import Segment
 
 class DataPipelineImplementation: DataPipelineInstance {
-    let moduleConfig: DataPipelineConfigOptions
-    let logger: Logger
+    private let moduleConfig: DataPipelineConfigOptions
+    private let logger: Logger
     let analytics: Analytics
 
+    private var globalDataStore: GlobalDataStore
+    // private let deviceAttributesProvider: DeviceAttributesProvider
+    private let dateUtil: DateUtil
+    private let deviceInfo: DeviceInfo
+
     init(diGraph: DIGraphShared, moduleConfig: DataPipelineConfigOptions) {
         self.moduleConfig = moduleConfig
         self.logger = diGraph.logger
         self.analytics = .init(configuration: moduleConfig.toSegmentConfiguration())
+
+        self.globalDataStore = diGraph.globalDataStore
+        // self.deviceAttributesProvider = diGraph.deviceAttributesProvider
+        self.dateUtil = diGraph.dateUtil
+        self.deviceInfo = diGraph.deviceInfo
+
+        initialize()
+    }
+
+    private func initialize() {
+        if let token = globalDataStore.pushDeviceToken {
+            // if the device token exists, pass it to the plugin to ensure device attributes are updated with each request
+            setDeviceToken(token)
```

Comment:
Wasn't able to test complete flow because Journey implementation is not yet complete. But testing with assumptions and looking at code, it looks like [segment is storing token only in memory](https://github.com/customerio/cdp-analytics-swift/blob/main/Sources/Segment/Plugins/DeviceToken.swift#L33), so we apparently need to pass it on every app launch.